jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppContext.java

```diff
@@ -958,8 +968,7 @@ else if (getResourceBase() != null)
      */
     public void setConfigurationClasses(String[] configurations)
     {
-        if (isStarted())
-            throw new IllegalStateException();
+        loadConfigurations();
```

Comment:
The behaviour now is different: previously setConfigurationClasses and setConfigurations would not cause the known configurations to be loaded first, meaning that the configurations were precisely only those contained in the set.